### PR TITLE
NTP-1244: Fix URL creation if a TP has forward slash in their name and tweak alerts

### PR DESCRIPTION
**Pull request by kieranohanlon**
## Context

Fix URL creation if a TP has forward slash in their name and tweak alerts

## Changes proposed in this pull request

A new TP has the name “Ravenspoint Ltd T/A Tutor Doctor Reading, Maidenhead, Wokingham“ the forward slash is causing an issue since we use this name to create the URL for the TP details.  So the / should be stripped out when used in the URL, otherwise the path is invalid.

We also spotted that if there is a missing logo we are not alerted (this is logged as info).  Since all TPs seem to have a logo we will log this as a warning (so will not stop upload, but we’ll be alerted), so we have visibility of the missing logo now that Tribal control the data maintenance for TPs.

Since the new spreadsheet successfully uploaded, we had an issue rolling back to the previous version.  This is because the old data could not be uploaded and was throwing an error, in case this is being done by accident.  To resolve this, we will set this as a warning, allowing old data to be uploaded, with a longer term solution being added to the backlog: https://dfedigital.atlassian.net/browse/NTP-1245

## Guidance to review

Run against new TP data with / and also make sure can rollback

## Link to Jira ticket

https://dfedigital.atlassian.net/browse/NTP-1244

## Things to check

- [x] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [x] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [ ] `dotnet format` has been run in the repository root
- [ ] Test coverage of new code is at least 80%
- [ ] All UI related acceptance criteria specified in the ticket have been added to the relevant [cypress test feature file(s)](/UI/cypress/e2e/)
- [ ] Unless explicitly mentioned in the ticket, all UI work should have been tested as working on desktop and mobile resolutions with JavaScript on and off
- [ ] Database migrations can be applied to the current codebase in main without causing exceptions
- [ ] Debug logging has been added after all logic decision points
- [ ] All [automated testing](/README.md#testing) including accessibility and security has been run against your code
- [ ] **PR deployment has been signed off by wider team**

### CODE REVIEW

File: Application/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -11,9 +11,17 @@ namespace Application.Extensions;
...
+    public static string? ToSeoUrl(this string? value, bool replaceForwardSlash = true)
     {
-        var seo = value?
+        if (string.IsNullOrEmpty(value))
+            return value;
+
+        var seo = value;
+
+        if (replaceForwardSlash)
+            seo = seo.Replace("/", "");
```
Comment (hasan3d) [inline Application/Extensions/StringExtensions.cs:22]: Are we expecting forward slash for the Seo Url?

Comment (hasan3d) [reply Application/Extensions/StringExtensions.cs:22]: If not we can incorporate the forward slash replacing logic with the code below.

Comment (kieranohanlon) [reply Application/Extensions/StringExtensions.cs:22]: I had originally put it within the `UrlUnsafeCharacters` - but on certain occasions we do need to leave the forward slashes, if part of full path.  See `ToSeoUrl_ReturnsKebabCase_WhenDirectory` and `SeoRouteConvention` within solution,

Comment (hasan3d) [reply Application/Extensions/StringExtensions.cs:22]: I see what you mean it's also used with directory, make sense.

**Review by hasan3d — APPROVED** [review]
lgtm

**Pull request merged**